java/memory/src/main/java/org/apache/arrow/memory/util/ArrowBufPointer.java

```diff
@@ -31,6 +34,15 @@
 
   private int length;
 
+  private int hashCode = 0;
```

Comment:
Could you make this a constant NULL_HASH_CODE  and use it here and below in hashCode

Reply:
Sure. Good suggestion.